pyflakes/messages.py

```diff
@@ -116,6 +116,22 @@ def __init__(self, filename, loc, name):
         self.message_args = (name,)
 
 
+class DuplicateDictionaryKey(Message):
+    message = 'dictionary key %s repeated with different values'
+
+    def __init__(self, filename, loc, key):
+        Message.__init__(self, filename, loc)
+        self.message_args = (key,)
+
+
+class DuplicateVariableDictionaryKey(Message):
```

Comment:
Agreed that the name probably needs improving. I'll do something with it momentarily.

Having two different classes for this allows the tests to be improved somewhat- so that we can tell a user that they have used duplicates of the string 'myvariable' vs the variable myvariable.

They're currently:
Duplicate<x>Variable: Any duplicated keys which are variables, including classes and functions, but NOT including any builtins (even in python2, special cased True,False,None).
This particular special case was introduced in regards to an earlier comment which (sensibly, I think?) pointed out that most people do not consider True, False, and None to be variables in python2 (even though they sort of are).

Duplicate<x>: Any duplicated keys which are builtins or other literals.

For the moment I'll just rename these to something stating Variable vs Literal and hopefully removing the disconnect. They can always be improved further as desired.